encode_test.go

```diff
@@ -101,6 +85,29 @@ func TestParseTs(t *testing.T) {
 	}
 }
 
+var timeErrorTests = []string{
+	"2001",
+	"2001-2-03",
+	"2001-02-3",
+	"2001-02-03 ",
+	"2001-02-03 04",
+	"2001-02-03 04:",
+	"2001-02-03 04:05",
+	"2001-02-03 04:05:",
+	"2001-02-03 04:05:6",
+	"2001-02-03 04:05:06.123 B",
```

Comment:
Are these values that are ever returned by Postgres for these types? If not, what is the performance impact of these changes?